browser/static/js/adjust.js

```diff
@@ -1,277 +1,376 @@
 
-  ctx.drawImage(preCompRaw, 0, 0, rawWidth, rawHeight);
+    this.contrastedRaw.src = this.canvas.toDataURL();
+  }
 
-  // add labels on top
-  ctx.save();
-  ctx.globalAlpha = 0.3;
-  ctx.drawImage(preCompSeg, 0, 0, rawWidth, rawHeight);
-  ctx.restore();
+  preCompAdjust(current_highlight, edit_mode, brush, mode) {
 
-  compositedImg.src = canvas.toDataURL();
-}
+    this.segLoaded = false;
 
-// apply outlines, transparent highlighting
-function postCompAdjust() {
-  let canvas = document.getElementById('hidden_seg_canvas');
-  let ctx = $('#hidden_seg_canvas').get(0).getContext('2d');
-  ctx.imageSmoothingEnabled = false;
+    // draw segImage so we can extract image data
+    this.ctx.clearRect(0, 0, this.width, this.height);
+    this.ctx.drawImage(this.segImage, 0, 0, this.width, this.height);
 
-  // draw seg_image so we can extract image data
-  ctx.clearRect(0, 0, rawWidth, rawHeight);
-  ctx.drawImage(compositedImg, 0, 0, rawWidth, rawHeight);
+    if (current_highlight) {
+      let segData = this.ctx.getImageData(0, 0, this.width, this.height);
+      let h1, h2;
 
-  // add outlines around conversion brush target/value
-  let imgData = ctx.getImageData(0, 0, rawWidth, rawHeight);
+      if (edit_mode) {
+        h1 = brush.value;
+        h2 = -1;
+      } else {
+        h1 = mode.highlighted_cell_one;
+        h2 = mode.highlighted_cell_two;
+      }
+
+      // highlight
+      this.preCompositeLabelMod(segData, h1, h2);
+      this.ctx.putImageData(segData, 0, 0);
+    }
 
-  let redOutline, r1, singleOutline, o1, outlineAll, translucent, t1, t2;
-  // red outline for conversion brush target
-  if (edit_mode && brush.conv && brush.target !== -1) {
-    redOutline = true;
-    r1 = brush.target;
+    // once this new src is loaded, displayed image will be rerendered
+    this.preCompSeg.src = this.canvas.toDataURL();
   }
-  if (edit_mode && brush.conv && brush.value !== -1) {
-    singleOutline = true;
-    o1 = brush.value;
+
+  // adjust raw further, pre-compositing (use to draw when labels hidden)
+  preCompRawAdjust() {
+
+    // further adjust raw image
+    this.ctx.clearRect(0, 0, this.width, this.height);
+    this.ctx.drawImage(this.contrastedRaw, 0, 0, this.width, this.height);
+    let rawData = this.ctx.getImageData(0, 0, this.width, this.height);
+    this._grayscale(rawData);
+    if (this.displayInvert) {
+      this._invert(rawData);
+    }
+    this.ctx.putImageData(rawData, 0, 0);
+
+    this.preCompRaw.src = this.canvas.toDataURL();
   }
 
-  postCompositeLabelMod(imgData, redOutline, r1, singleOutline, o1,
-                        outlineAll, translucent, t1, t2);
+  // composite annotations on top of adjusted raw image
+  compositeImages() {
 
-  ctx.putImageData(imgData, 0, 0);
+    this.ctx.drawImage(this.preCompRaw, 0, 0, this.width, this.height);
 
-  postCompImg.src = canvas.toDataURL();
-}
+    // add labels on top
+    this.ctx.save();
+    this.ctx.globalAlpha = 0.3;
```

Comment:
What is this number? Should it be paramterized?

Reply:
That determines how transparent labels should be when they're drawn on top of the raw image to make the pixel-editing overlay image. Parameterizing it sounds good, especially since it would be nice (eventually) to have it be an adjustable value.